app/frontend/types.ts

```diff
@@ -55,7 +55,19 @@ export type _XPubKey = {
   xpubHex: HexString
 }
 
-export type AuthMethod = '' | 'hw-wallet' | 'mnemonic' // TODO
+export enum AuthMethodEnum {
+  INITIAL,
```

Comment:
this looks to me like an abuse of an enum that is meant to represent actual AuthMethods. If I understand correctly, this represents the fact that no AuthMethod has been selected yet which is kind of counterintuitive to encapsulate within the AuthMethod itself and can expose logic handling different actual AuthMethods to deal with this weird case.

I would find it more natural to not have this "initial" type at all and instead have the logic that expects that to have an interface expecting an optional type (?AuthMethodEnum) and logic that is expected to handle valid AuthMethods to expect a non-optional type. And the initial AuthMethod in the state would just be null, idiomatically representing the fact that no auth method has been selected